src/Functions/FunctionsConversion.h

```diff
@@ -990,20 +990,11 @@ struct ConvertImpl<FromDataType, DataTypeString, Name, ConvertDefaultBehaviorTag
         {
             ColumnUInt8::MutablePtr null_map = copyNullMap(arguments[0].column);
 
-            const auto & col_with_type_and_name =  columnGetNested(arguments[0]);
+            const auto & col_with_type_and_name = columnGetNested(arguments[0]);
             const auto & type = static_cast<const FromDataType &>(*col_with_type_and_name.type);
 
             const DateLUTImpl * time_zone = nullptr;
 
-            if constexpr (std::is_same_v<FromDataType, DataTypeDate> || std::is_same_v<FromDataType, DataTypeDate32>)
```

Comment:
we could add assert like `!IsDataTypeDateOrDateTime<FromDataType>` to make sure this code is never actually needed

Reply:
[Here](https://github.com/yariks5s/ClickHouse/blob/66479b69660e999c0f0a58a7aa976e1714869fbb/src/Functions/FunctionsConversion.h#L888) we have check regarding `IsDataTypeDateOrDateTime`, so as far as I concerned this check is useless, maybe I am wrong.

Reply:
Nope, you're right